### PR TITLE
Feature/recipe screen UI

### PR DESCRIPTION
**Pull request by anaselmadhi**
# Add recipe screen UI
### What has been changed?

This PR introduces the user interface for the liked recipes screen allowing users to view their favorite recipes in a card-based layout. Each card displays details such as the recipe title, rating, preparation time, and cost. Additionally, a search bar and a bottom navigation menu are included to enhance the user experience.

### Why was this change made?

The liked recipes screen enhances the app by offering a dedicated view for liked recipes, making it easier for users to revisit and manage their favorite meals. This feature aligns with our goal of creating a more intuitive and engaging user interface, enabling efficient recipe browsing and a more personalized app experience.

I have also added two dependencies: 
    implementation("io.coil-kt:coil-compose:2.0.0")
    implementation("androidx.compose.material:material-icons-extended:1.5.0")

### Search Bar
Currently, the search bar only provides the graphical interface. The functionality to filter recipes based on the search input is not yet implemented in the ViewModel. This feature is planned for a future update, where the search bar will actively filter recipes as users type.

### Like/Dislike Functionality
At this stage, the "like" functionality only switches between a filled and an empty heart icon to indicate liked and unliked recipes. The ViewModel does not yet support the functionality to remove unliked recipes from the list, but this will be addressed in an upcoming update. In addition, a delay will be added to automatically remove unliked recipes (with empty icons) from the list of saved recipes after a certain period.

### Rating, Preparation Time, and Price
Currently, data for the rating, preparation time, and price are not available in the recipe database. For now, we are using random values to represent these fields in the UI. In future updates, once these fields are integrated into the database, the values will reflect actual data.

## Checklist
- [x] Tests added.
- [x] Documentation updated.
- [ ] All CI checks passed.
- [x] Linked issue/feature (if applicable): #88

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -148,6 +148,10 @@ dependencies {
     implementation(composeBom)
     globalTestImplementation(composeBom)
 
+    // ecrire aussi sur lib version pour meileur format
+    implementation("io.coil-kt:coil-compose:2.0.0")
+    implementation("androidx.compose.material:material-icons-extended:1.5.0")
```
Comment (RagdollGoDown) [inline app/build.gradle.kts:153]: Even though these dependencies are only for icons and minor compose additions, next time do a separate PR for the dependencies.

File: app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt [hunk 2]
```diff
@@ -1,15 +1,239 @@
...
+                      Modifier.padding(start = 16.dp, top = 8.dp, bottom = 16.dp)
+                          .height(64.dp)
+                          .width(329.dp)
+                          .testTag("SearchButton")) {
+                    Row(modifier = Modifier.fillMaxWidth()) {
+                      Icon(
+                          modifier = Modifier.testTag("searchButtonIcon"),
+                          imageVector = Icons.Filled.Search,
+                          contentDescription = "Search",
+                          tint = Color.Gray)
```
Comment (RagdollGoDown) [inline app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt:183]: For colors, like the tint here or the color of the background, please use the ones given in the theme.

File: app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt [hunk 3]
```diff
@@ -1,15 +1,239 @@
...
+import java.util.Random
 
+/**
+ * Recipe card composable that displays a recipe.
+ *
+ * @param recipe recipe to display
+ */
 @Composable
-fun RecipeScreen() {
+fun RecipeCard(recipe: Recipe) {
```
Comment (RagdollGoDown) [inline app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt:57]: This gives us the result we want, but it would be better if it were separated into smaller composable functions ex: `RecipeTitle`. This would greatly help in making the code more readable.

File: app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt [hunk 4]
```diff
@@ -1,15 +1,239 @@
...
+            }
+      },
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { destination -> navigationActions.navigateTo(destination) },
+            tabList = LIST_TOP_LEVEL_DESTINATIONS,
+            selectedItem = navigationActions.currentRoute())
+      },
+  ) {
+    Column(
```
Comment (RagdollGoDown) [inline app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt:205]: We need to use Lazycolumn instead of Column to keep things optimized. Something like: 
```Kotlin
LazyColumn(
    modifier = Modifier.padding(it).fillMaxWidth(),
    verticalArrangement = Arrangement.Center,
    horizontalAlignment = Alignment.CenterHorizontally
) {
    items(recipesList) { recipe ->
        RecipeCard(recipe)
    }
}

**Review by RagdollGoDown — CHANGES_REQUESTED** [review]
Well done on the progress for the liked recipe list! It's nice that users will be able to see their favourite recipes! Although, overall the code is good, there are a few improvements that can be made in terms of readability, maintainability and performance.

**Review by RagdollGoDown — APPROVED** [review]

File: app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt (hunk 2 above)
Comment (anaselmadhi) [reply app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt]: I agree with that. I'll make sure to use the colors provided in the theme for things like tints and background colors.

File: app/build.gradle.kts (hunk 1 above)
Comment (anaselmadhi) [reply app/build.gradle.kts:153]: I thought that making a PR for just these two implementations wouldn't have been very useful, but it's true that it's important to do so to properly track the project's progress and get approval from other team members for any additions. I'll make sure to create separate PRs for each change next time!

File: app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt (hunk 3 above)
Comment (anaselmadhi) [reply app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt:61]: I completely agree. To improve code readability, I’ll make sure to refactor it into smaller, more manageable composable functions.

File: app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt (hunk 4 above)
Comment (anaselmadhi) [reply app/src/main/java/com/android/sample/ui/testScreens/RecipeScreen.kt]: That's a great idea! I'll implement that to keep the layout optimized. Thanks for the suggestion!
